### PR TITLE
add REST API view model

### PR DESCRIPTION
**Pull request by filariow**
With this PR the separation of the REST API's ViewModel from the Operator's CRD is complete.

The operator will work with `InternalWorkspaces.workspaces.konflux.io`, whereas the communication between the user and the REST API server will use the virtual resource `Workspaces.workspaces.konflux.io`.

The REST API has the responsibility of translating `Workspaces` to `InternalWorkspaces` and viceversa.

### CODE REVIEW

File: server/core/workspace/workspace_update.go [hunk 1]
```diff
@@ -50,6 +51,7 @@ func (h *UpdateWorkspaceHandler) Handle(ctx context.Context, query UpdateWorkspa
 
 	// data access
 	w := query.Workspace.DeepCopy()
+	log.FromContext(ctx).Info("updating workspace", "workspace", w)
```
Comment (sadlerap) [inline server/core/workspace/workspace_update.go:54]: I suspect dumping the entire workspace is going to pollute our logs quite a bit.  Maybe we could prune it down a bit?

**Review by sadlerap — COMMENTED** [review]
one thought, otherwise lgtm so far

File: server/core/workspace/workspace_update.go (hunk 1 above)
Comment (filariow) [reply server/core/workspace/workspace_update.go:54]: good catch, I'd say we can use Debug here. WDYT?

Comment (sadlerap) [reply server/core/workspace/workspace_update.go:54]: lgtm

Comment (filariow) [reply server/core/workspace/workspace_update.go]: (done in cf201edff9b2211bd62fcb2fdeb4d17547619b89)

**Review by sadlerap — APPROVED** [review]

**Pull request merged**